packages/core/src/tools/utils/urlPolyfill.ts

```diff
@@ -18,40 +18,43 @@ export function getPathName(url: string) {
 }
 
 export function buildUrl(url: string, base?: string) {
-  const supportedURL = getSupportedUrl()
-  if (supportedURL) {
-    try {
-      return base !== undefined ? new supportedURL(url, base) : new supportedURL(url)
-    } catch (error) {
-      throw new Error(`Failed to construct URL: ${String(error)} ${jsonStringify({ url, base })!}`)
-    }
-  }
-  if (base === undefined && !/:/.test(url)) {
-    throw new Error(`Invalid URL: '${url}'`)
-  }
-  let doc = document
-  const anchorElement = doc.createElement('a')
-  if (base !== undefined) {
-    doc = document.implementation.createHTMLDocument('')
-    const baseElement = doc.createElement('base')
-    baseElement.href = base
-    doc.head.appendChild(baseElement)
-    doc.body.appendChild(anchorElement)
+  const nativeURL = getNativeURL()
+  const URLConstructor = nativeURL || URL
+
+  try {
+    return base !== undefined ? new URLConstructor(url, base) : new URLConstructor(url)
+  } catch (error) {
+    throw new Error(`Failed to construct URL: ${String(error)} ${jsonStringify({ url, base })!}`)
   }
-  anchorElement.href = url
-  return anchorElement
 }
 
-const originalURL = URL
-let isURLSupported: boolean | undefined
-function getSupportedUrl(): typeof URL | undefined {
-  if (isURLSupported === undefined) {
-    try {
-      const url = new originalURL('http://test/path')
-      isURLSupported = url.href === 'http://test/path'
-    } catch {
-      isURLSupported = false
+/**
+ * Get native URL constructor from a clean iframe
+ * This avoids polyfill issues by getting the native implementation from a fresh iframe context
+ * Falls back to the original URL constructor if iframe approach fails
+ */
+let cachedNativeURL: typeof URL | undefined
+export function getNativeURL(): typeof URL | undefined {
+  if (cachedNativeURL !== undefined) {
+    return cachedNativeURL
+  }
+  try {
+    const iframe = document.createElement('iframe')
```

Comment:
Two nits here:
1. It'd be a good idea to set `iframe.style.display = 'none'`, so that the iframe will not be rendered if anything goes wrong. (Right now, if something goes wrong, the iframe would be quite visible!)
2. It'd be a good idea to remove the iframe from `document.body` in a `finally` block, so that it's guaranteed to happen even if an exception is thrown.

So, to start with, you might want to make these changes:

```suggestion
  }
  let iframe: HTMLIFrameElement
  try {
    iframe = document.createElement('iframe')
    iframe.style.display = 'none'
```